tests/integration/targets/java_cert/tasks/state_change.yml

```diff
@@ -4,19 +4,89 @@
   args:
     creates: "{{ test_key_path }}"
 
-- name: Create the test keystore
-  java_keystore:
-    name: placeholder
-    dest: "{{ test_keystore2_path }}"
-    password: "{{ test_keystore2_password }}"
-    private_key: "{{ lookup('file', '{{ test_key_path }}') }}"
-    certificate: "{{ lookup('file', '{{ test_cert_path }}') }}"
-
 - name: Generate the self signed cert we will use for testing
   command: openssl req -x509 -newkey rsa:4096 -keyout '{{ test_key2_path }}' -out '{{ test_cert2_path }}' -days 365 -nodes -subj '/CN=localhost'
   args:
     creates: "{{ test_key2_path }}"
 
+- name: Create the pkcs12 archive from the test x509 cert
+  command: >
+    openssl pkcs12
```

Comment:
Why not use the `openssl_pkcs12` module?

Reply:
I think this comment can be generally applied to this module :). It is managing three sources for certificates; x509 stores, pkcs12 bundles, and most confusingly directly from a SSL/TLS enabled server. There are better means of retrieving certificates with existing modules that are maintained just for that purpose. And in regards to importing server-side certs directly from the server people are free to do what they want, but it defeats the purpose of maintaining a truststore. Not saying any action is required at the moment, but I think any unique functionality of this module could be stripped out into something new that plays well with `community.crypto` and `java_keystore`.

Reply:
> Why not use the `openssl_pkcs12` module?

I would not introduce a dependency on community.crypto for the test, should I ? anyway, it's only a migration of these tasks to another location in the file, it's not my creation and it works well as far as I know :)

Reply:
The dependency for integration testing is already there: https://github.com/ansible-collections/community.general/blob/main/tests/requirements.yml#L3

But yeah, let's not do too much in this PR :)